app/ydoc-shared/src/ast/mutableModule.ts

```diff
@@ -272,23 +271,28 @@ export class MutableModule implements Module {
         ])
         updateBuilder.updateFields(id, changes)
       } else if (event.target.parent.parent === this.nodes) {
-        // Updates to fields of a metadata object within a node.
+        // Updates to fields of an object within a node.
         const id = event.target.parent.get('id')
         DEV: assertAstId(id)
         const node = this.nodes.get(id)
         if (!node) continue
         const metadata = node.get('metadata') as unknown as Map<string, unknown>
-        const changes: (readonly [string, unknown])[] = Array.from(event.changes.keys, ([key]) => [
-          key,
-          metadata.get(key as any),
-        ])
-        updateBuilder.updateMetadata(id, changes)
-      } else if (event.target.parent.parent.parent === this.nodes) {
-        // Updates to some specific widget's metadata
-        const id = event.target.parent.parent.get('id')
-        assertAstId(id)
-        if (!this.nodes.get(id)) continue
-        updateBuilder.updateWidgets(id)
+        if (event.target === metadata) {
+          const changes: (readonly [string, unknown])[] = Array.from(
+            event.changes.keys,
+            ([key]) => [key, metadata.get(key as any)],
+          )
+          updateBuilder.updateMetadata(id, changes)
+        } else if (event.target.parent === metadata) {
+          // Updates to some specific widget's metadata
+          const id = event.target.parent.parent.get('id')
```

Comment:
If we needed to add a check above, that `event.target` is metadata, then maybe we should also check here, that event.target is `widget` of `metadata`?